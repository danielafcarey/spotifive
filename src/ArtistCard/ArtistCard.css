.artist-card {
  display: flex;
  flex-direction: row;
  align-items: center; }
  .artist-card:hover {
    cursor: pointer; }
  .artist-card img {
    height: 50px;
    border-radius: 25px; }
  .artist-card p {
    text-align: left;
    padding: 0 15px; }
