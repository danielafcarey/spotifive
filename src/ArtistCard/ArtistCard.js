import React from 'react';

const ArtistCard = ({ name, id, selectArtist }) => {
  return (
    <div 
      onClick={ () => selectArtist(id) }
    >
      <p>{ name }</p>    
    </div>
  )
}

export default ArtistCard;
