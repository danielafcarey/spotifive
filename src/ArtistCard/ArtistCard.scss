.artist-card {
  display: flex;
  flex-direction: row;
  align-items: center;

  &:hover {
    cursor: pointer;
    background-color: #272727;
  }

  img {
    height: 50px;
    padding: 10px;
    border-radius: 50%;
  }

  p {
    text-align: left;
    padding: 0 15px;
    font-size: 25px;
  }
}
