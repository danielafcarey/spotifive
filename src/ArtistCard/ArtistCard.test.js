import React from 'react';
import { shallow } from 'enzyme';
import ArtistCard from './ArtistCard';

describe('ArtistCard', () => {

  it('matches the snapshot if the image is null', () => {
    const mockProps = {
      name: 'Daniela',
      id: 'abc',
      image: null,
      selectArtist: jest.fn()
    }
    const wrapper = shallow(<ArtistCard { ...mockProps }/>);

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot if the image is not null', () => {
    const mockProps = {
      name: 'Daniela',
      id: 'abc',
      image: 'image',
      selectArtist: jest.fn()
    }
    const wrapper = shallow(<ArtistCard { ...mockProps }/>);

    expect(wrapper).toMatchSnapshot();
  })
})
