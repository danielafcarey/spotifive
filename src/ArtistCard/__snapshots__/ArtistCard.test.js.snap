// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard matches the snapshot if the image is not null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistCard
    id="abc"
    image="image"
    name="Daniela"
    selectArtist={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Daniela"
          src="image"
/>,
        <p>
          Daniela
</p>,
      ],
      "className": "artist-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Daniela",
          "src": "image",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Daniela",
        },
        "ref": null,
        "rendered": "Daniela",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Daniela"
            src="image"
/>,
          <p>
            Daniela
</p>,
        ],
        "className": "artist-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Daniela",
            "src": "image",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Daniela",
          },
          "ref": null,
          "rendered": "Daniela",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ArtistCard matches the snapshot if the image is null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistCard
    id="abc"
    image={null}
    name="Daniela"
    selectArtist={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Daniela"
          src="logo-grey.png"
/>,
        <p>
          Daniela
</p>,
      ],
      "className": "artist-card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Daniela",
          "src": "logo-grey.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Daniela",
        },
        "ref": null,
        "rendered": "Daniela",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Daniela"
            src="logo-grey.png"
/>,
          <p>
            Daniela
</p>,
        ],
        "className": "artist-card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Daniela",
            "src": "logo-grey.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Daniela",
          },
          "ref": null,
          "rendered": "Daniela",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
